### PR TITLE
Request validation

### PR DESCRIPTION
**Pull request by rgreinho**
Add the ability to parse the POST request received and execute a command if the validation is successful.
Add the ability to read a `Hookletfile` that contains the command to execute and the data to validate.
Add some UnitTests.
Update the instructions.

### CODE REVIEW

File: hooklet/hooklet.py [hunk 1]
```diff
@@ -1,12 +1,137 @@
...
+def read_hookletfile():
+    """Opens and parses the Hookletfile.
+
+    :return: a JSON object representing the Hookletfile.
+    """
+    # Open the Hookletfile.
+    hookletfile = open('Hookletfile', 'r')
+
+    # Parse it.
+    return yaml.safe_load(hookletfile)
```
Comment (fridiculous) [inline hooklet/hooklet.py]: close the file

File: hooklet/tests/test_hooklet.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
...https://github.com/rgreinho/hooklet","homepage":null,"size":0,"stargazers_count":0,"watchers_count":0,"language":"Python","has_issues":true,"has_downloads":true,"has_wiki":true,"has_pages":false,"forks_count":0,"mirror_url":null,"open_issues_count":0,"forks":0,"open_issues":0,"watchers":0,"default_branch":"master"},"sender":{"login":"rgreinho","id":6969134,"avatar_url":"https://avatars.githubusercontent.com/u/6969134?v=3","gravatar_id":"","url":"https://api.github.com/users/rgreinho","html_url":"https://github.com/rgreinho","followers_url":"https://api.github.com/users/rgreinho/followers","following_url":"https://api.github.com/users/rgreinho/following{/other_user}","gists_url":"https://api.github.com/users/rgreinho/gists{/gist_id}","starred_url":"https://api.github.com/users/rgreinho/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/rgreinho/subscriptions","organizations_url":"https://api.github.com/users/rgreinho/orgs","repos_url":"https://api.github.com/users/rgreinho/repos","events_url":"https://api.github.com/users/rgreinho/events{/privacy}","received_events_url":"https://api.github.com/users/rgreinho/received_events","type":"User","site_admin":false}}'
```
Comment (fridiculous) [inline hooklet/tests/test_hooklet.py]: please break this up. it shouldn't be 1 line.

you can also save it a new file or as a mock object

File: hooklet/hooklet.py [hunk 3]
```diff
@@ -1,12 +1,137 @@
...
-@app.route("/")
-def hello():
-    return "Hello world!"
+@app.route("/git/pull/war", methods=['POST'])
+def git_pull_repository():
+    """Pulls the war repository.
+
+    If the request payload is valid, then the command specified in the Hookletfile will be executed.
+
+    WARNING: This command is ABSOLUTELY unsafe as ANY bash script can be executed from it!
```
Comment (fridiculous) [inline hooklet/hooklet.py]: why don't you just hardcode the git pull logic and send a repo as a param. it would make it safe, no?

Comment (rgreinho) [inline hooklet/hooklet.py]: I thought about it, but the command is different for the public and private repos.

**Pull request closed without merging**